### ISSUE TITLE
El Capitan + Dryscrape

### ISSUE THREAD

**Issue by polyrhythmatic** [niklasb/dryscrape#44]
I've been stuck on this issue for a while and I'm wondering if this is a compatibility issue with El Capitan. Running a script with dryscrape yields this error

```
Traceback (most recent call last):
  File "concat.py", line 12, in <module>
    dryscrape.start_xvfb()
  File "/Users/sethkranzler/Development/cornell_project/cornell_scrape/env/lib/python2.7/site-packages/dryscrape/xvfb.py", line 9, in start_xvfb
    xvfb.start()
  File "/Users/sethkranzler/Development/cornell_project/cornell_scrape/env/lib/python2.7/site-packages/xvfbwrapper.py", line 53, in start
    stderr=fnull)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

This script was working perfectly fine before I upgraded and works fine on my raspberry pi 2. 

Any idea what could be causing this?

**Comment by MadsRC** [comment]
Seems to be the same as this problem: https://github.com/niklasb/dryscrape/issues/15

I got the very same problem on a OSX 10.11 El Capitan.
Also got the very same problem on Arch Linux.

Install qt5 from repo's, and pip install lxml, webkit-server and dryscrape

**Comment by niklasb** [comment]
Is this still an issue? If you do `pip install webkit-server`, it should install the binary and then you shouldn't get this error.

**Comment by niklasb** [comment]
@MadsRC So what you describe did in fact lead to a working dryscrape?

**Comment by MadsRC** [comment]
Unfortunately no, I ended up using another scraper for my project.

Later, when I'm off work, I'll try again from scratch and see if I can get around it.

**Comment by niklasb** [comment]
@MadsRC Fair enough :) As soon as I get home I will try to install on my Mac to see if I can somehow reproduce this.

**Comment by MRHarrison** [comment]
I'm still having this issue.  Did anybody find a solution. Thanks.

**Comment by polyrhythmatic** [comment]
I'm sorry for the delay, this issue forced me to find another solution to my problem (which actually wound up being a better one :+1: )

I'll try to find some time this weekend to see if there is a process to solve this.

**Comment by MRHarrison** [comment]
@polyrhythmatic Thanks for taking a look again, could you please let me what alternative you found.  I'm looking for a lightweight option to scrape all pages, JS rendered or not, but the only other reliable option I know of is heavy, which is a chromdriver and selenium.  (I prefer chromedriver over phantomjs).  Thanks.

**Comment by polyrhythmatic** [comment]
@MRHarrison I ended up realizing I could reconstruct the URL I was after as opposed to running js to get it. So this breaking turned out to be a good thing in a way...

I'm curious why a utility like beautiful soup wouldn't work for your needs. Have you checked it out?

**Comment by polyrhythmatic** [comment]
@MadsRC I tried this, though I used brew install qt5 and it did not work. Uninstalled lxml, webkit-server, and dryscrape and reinstalled.

**Comment by jmarovt** [comment]
Since this is a related El Capitan issue, I'm posting the traceback here. Tried different python version, qt, qt5, reinstalling webkit_server, lxml, etc. Would love some help.

`Traceback (most recent call last):
  File "atsscan.py", line 27, in <module>
    dryscrape.start_xvfb()
  File "/Users/jmarovt/.pyenv/versions/3.3.0/lib/python3.3/site-packages/dryscrape/xvfb.py", line 8, in start_xvfb
    xvfb = Xvfb()
  File "/Users/jmarovt/.pyenv/versions/3.3.0/lib/python3.3/site-packages/xvfbwrapper.py", line 27, in __init__
    raise EnvironmentError(msg)
OSError: Can not find Xvfb. Please install it and try again.`

**Comment by niklasb** [comment]
@jmarovt Do you actually have xvfb installed?

**Comment by niklasb** [comment]
@jmarovt Why do you use `start_xvfb` on Mac OS X? This is only needed on a X-less Linux server

**Comment by niklasb** [comment]
@polyrhythmatic Same for you. I should probably update the examples to make this clearer

**Comment by niklasb** [comment]
fixed in f889c7fc9c1e9658ce03e3327baf2f7135d0b013

**Comment by 0z4ck** [comment]
The error of @jmarovt is different from ours.

I'm using webkit-server dowlnloaded by 'easy_install' because the package which is supposed to install by 'pip install webkit-server' has an installation issue(showed bellow).

and,
I'm sure 
 the webkit-server installed by easy_install contains a bad statement in calling subprocess module.


```
Installing collected packages: webkit-server
  Running setup.py install for webkit-server ... error
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-JUINvU/webkit-server/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-KQbgQA-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    sh: qmake: command not found
    error: src/webkit_server: No such file or directory
    
    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-JUINvU/webkit-server/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-KQbgQA-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-JUINvU/webkit-server/
```
